nvflare/private/fed/server/job_validator.py

```diff
@@ -0,0 +1,170 @@
+
+    def validate(self, job_name: str, job_data: bytes) -> Tuple[bool, str, dict]:
+        """Validate job
+
+        Args:
+            job_name (str): Job name
+            job_data (bytes): Job ZIP data
+
+        Returns:
+            Tuple[bool, str, dict]: (is_valid, error_message, meta)
+        """
+
+        meta = {}
+        try:
+            with ZipFile(BytesIO(job_data), "r") as zf:
+                meta_file = job_name + "/meta.json"
+                meta_data = zf.read(meta_file)
+                meta = json.loads(meta_data)
+                if JobMetaKey.JOB_FOLDER_NAME not in meta:
+                    meta[JobMetaKey.JOB_FOLDER_NAME.value] = job_name
+
+                study_name = meta.get(JobMetaKey.STUDY_NAME.value)
+                self._validate_study(study_name, meta)
+                self._validate_deployment(meta, zf)
+
+                # Validating resource_spec
+                resource_spec = meta.get(JobMetaKey.RESOURCE_SPEC.value)
+                if resource_spec and not isinstance(resource_spec, dict):
+                    raise ValueError(f"Invalid resource_spec for job {job_name}")
+
+        except ValueError as e:
+            return False, str(e), meta
+
+        return True, "", meta
+
+    def _validate_study(self, study_name: str, meta: dict) -> None:
+
+        # Default or empty study is always valid
+        if not study_name or study_name == Study.DEFAULT_STUDY_NAME:
+            return
+
+        study = self.study_manager.get_study(study_name, self.fl_ctx)
+        if not study:
+            raise ValueError(f"Study {study_name} does not exist")
+
+        # Validating all clients are participants of the study
+        all_clients = self._get_all_clients(meta)
+        if study.participating_clients:
+            participants = set(study.participating_clients)
+            if not all_clients.issubset(participants):
+                diff = all_clients - participants
+                raise ValueError("Following clients are not participants of the study {}: {}".format(study_name, diff))
+
+        # Validating study validity period
+        current_time = time.time()
+        if study.start_time and study.start_time > current_time:
+            raise ValueError(
+                f"Study {study_name} has not started yet. "
+                f"The start time is {datetime.utcfromtimestamp(study.start_time).isoformat()}"
+            )
+
+        if study.end_time and study.end_time < current_time:
+            raise ValueError(
+                f"Study {study_name} has ended. "
+                f"The end time is {datetime.utcfromtimestamp(study.end_time).isoformat()}"
+            )
+
+    def _validate_deployment(self, meta: dict, zip_file: ZipFile) -> None:
+
+        job_name = meta.get(JobMetaKey.JOB_FOLDER_NAME)
+        deploy_map = meta.get(JobMetaKey.DEPLOY_MAP)
+        if not deploy_map:
+            raise ValueError(f"deploy_map is missing in meta.json for job {job_name}")
```

Comment:
how to handle deploy_map less situation

Reply:
deploy_map is a required filed in meta.json. It can contain empty site list but the map itself can't be empty.